src/server/game/Movement/MovementGenerators/PointMovementGenerator.cpp

```diff
@@ -142,8 +142,6 @@ void EffectMovementGenerator::Finalize(Unit* unit)
     {
         if (Unit* victim = unit->GetVictim())
             unit->GetMotionMaster()->MoveChase(victim);
-        else
-            unit->GetMotionMaster()->Initialize();
```

Comment:
MotionMaster::Initialize drops all other movement generators and initializes default one (set in DB for creatures). This behavior is completely wrong, there is no reason to drop all other movement generators when this one is finalized.